### ISSUE TITLE
largo de byte, para pasar de int a byte

### ISSUE THREAD

**Issue by vice-rosas99** [IIC2233/Syllabus#343]
### Prerrequisitos
(Marcar colocando una X entre los corchetes los ítems que ya hiciste, así: "[X]")

* [X] Leí las reglas del foro (https://github.com/IIC2233/syllabus/issues/1)
* [X] Busqué en las *issues* si ya preguntaron mi duda y no encontré nada parecido (https://github.com/IIC2233/syllabus/issues)
* [X] Revisé el compilado de dudas de la tarea y no encontré una *issue* similar a la mía (https://github.com/IIC2233/syllabus/issues/293)
* [X] Mi duda no se trata sobre una librería, *built-in* o mala práctica, ya que eso se pregunta en la *issue* creada para ello.
* [X] Mi duda no se trata de un tema administrativo o personal, ya que en ese caso debo contactar a mi profe, al Jefe de Bienestar o al correo del curso (https://iic2233.github.io/contacto/)
* [X] Utilizaré un título descriptivo y llenaré correctamente esta plantilla
* [X] De ser necesario, solo colocaré código simple que permita explicar mi problema o duda, ya que compartir código de esta evaluación puede ser considerado copia

### Duda
me surgió una duda sobre el tamaño de los bytes ya que en mi desencriptador de mesajes estoy trabajando con un metodo built in de python, para pasar 'int' a byte, pero es necesario saber el tamaño del byte, probando con distintos ints me doy cuenta de que no todos los números pueden ser representados con un byte de tamaño 1.
**Por esto pregunto su puedo asumir que todos los bytes en los mensajes se representan con un int que al mismo tiempo es representado por un byte de tamaño 1?

**Comment by tocococa** [comment]
Hola @vice-rosas99, un byte siempre ocupa 8 bits, los objetos `int` en Python tienen un método que te permite castearlos a una cantidad de bytes fija o viceversa, `to_bytes()` y `from_bytes`. Con un byte puedes guardar hasta 255, con 4 (que es lo que se pide en el enunciado) es un número tan grande que no tienes que preocuparte que no quepa su valor.

**Comment by vice-rosas99** [comment]
claro, preguntaba porque estoy usando el método to_bytes() pero necesito un argumento lenght, que sería el tamaño del byte a dar, por eso pregunto si siempre tendré que dar ese valor como 1

**Comment by tocococa** [comment]
El argumento `length` de ese método no indica el largo del byte, si no el número de bytes, y ahí tienes que colocar el número de bytes que se indica en el enunciado.

**Comment by vice-rosas99** [comment]
Para ser más claro, es por el header que indica el largo de lo que voy a enviar

**Comment by vice-rosas99** [comment]
Mmm es que no se como explicarlo, pq no es para la parte de poner el header y todo eso, lo estoy usando en la desencriptación
si algun ayudante pudiera mirar mi archivo de codificación_de_mensajes.py

**Comment by vice-rosas99** [comment]
aver lo que pasa es que yo recorro un bytearray en un for, y al ir sacando cada parte del bytearray, no me da objetos de bytes, sino que me da ints, por lo que debo convertir esos ints en bytes:
ej: 
a = bytearray(mensaje) # una coleccion de bytes
for i in range(0, len(a):
    elemento = a[i]
en este caso elemento es de tipo int, y necesito que sea byte
por eso pregunto por lo que debería poner en length, ya que para devolverlo a byte pongo:
   elemento.to_byes( length, byteorder), que hasta ahora e trabajado como 1 y 'little' respectivamente

**Comment by tocococa** [comment]
Si lo que muestras en tu snippet es correcto, no se deberían estar casteando a `int`, lo que puede ocurrir, es que si haces `print()` del byte, este se imprima como el `char` al que corresponde, y por eso estás viendo números, pero también podría ser otro carácter. Ahora si `isinstance(elemento, int)` retorna `True` puede ser que los estés casteando implícitamente a 'int' en alguna parte.
